coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -781,11 +779,18 @@ def flatten(context, node):
         raise ValueError(
             "Start must be before end. (start, end) == ({}, {})".format(start, end_val)
         )
-    for dim in dims[start : end + 1]:
-        total *= dim
-    dims = dims[:start] + [total] + dims[end + 1 :]
+    x_shape = mb.shape(x=x)
+
+    indices1 = mb.range_1d(start=0, end=start, step=1)
```

Comment:
This could be replaced by `mb.slice_by_index` or `mb.slice_by_size`.
I would suggest using them to simplify the implementation